pkg/controller/customdomain/customdomain_utils.go

```diff
@@ -228,3 +229,22 @@ func randSeq(n int) string {
 	}
 	return string(b)
 }
+
+// sameData compares the data in two secrets to determine if they contain the same info
+func sameData(a, b *corev1.Secret) bool {
```

Comment:
Is there a reason we can't just use [DeepEqual](https://pkg.go.dev/reflect#DeepEqual) instead?